lib/sles4sap_publiccloud.pm

```diff
@@ -178,7 +178,8 @@ sub sles4sap_cleanup {
     return 0 if ($args{cleanup_called});
 
     # If there's an open ssh connection to the VMs, return to host console first
-    select_host_console(force => 1);
+    # Useful for tests that use ssh_interactive modules
+    select_host_console(force => 1) if length get_var('MR_TEST','');
```

Comment:
```suggestion
    select_host_console(force => 1) if (get_var('MR_TEST',''));
```

It would also probably help with the pipeline failure.

Reply:
Wanted to guard against `0`... but MR_TEST has full string values anyway. Done

Reply:
> Wanted to guard against `0`... but MR_TEST has full string values anyway. Done

Hmm ... good point.

That would cause the test to fail on https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/sles4sap/saptune/mr_test_run.pm#L863 as `0` is not a valid `mr_test` test, but by this time in the case of cloud tests, the infrastructure is probably already deployed. It would run the `post_fail_hook` from `sles4sap/saptune/mr_test_run`, which will not destroy the cloud infrastructure, but then the next 2 modules (`ssh_interactive_end` and `qesap_cleanup`) would run ... so you may be onto something trying to catch `0` here.

Reply:
This is what it looks like when `mr_test_run` fails: https://openqa.suse.de/tests/18030082#step/1_saptune_notes/66

If `qesap_cleanup` calls this function (I think it does), then you're right in catching a `0` string there.